[yangyi-zhu] iP

yangyi-zhu opened the pull request.

yangyi-zhu committed:
Added greeting and echo
Renamed chatbot

yangyi-zhu committed:
Implemented ability to store user input and display list upon request

yangyi-zhu committed:
Implemented marking a task as done and vice versa

yangyi-zhu committed:
Added support for tracking three types of tasks

yangyi-zhu committed:
Restructured printed messages for enhanced readability

LEESY02 commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,94 @@
+    private static final Task[] list = new Task[100];
+
+    /**
+    public static void echo(String msg) {
```
The commented code seems to be a snippet of code from a previous Level, perhaps it could be removed as it does not help the readers understand what is going on with your code.

LEESY02 commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,94 @@
+    public static void checkInput(String msg) {
+        if (msg.equals("list")) {
+            System.out.println(NL + DIVIDER + NL + "Tasks:");
+            for (int i = 0; i < count; i++) {
```
Maybe you could refactor each of the if cases into a method

Instead of using multiple lines of else if, perhaps having a single case() switch will be easier to read as well

LEESY02 replied:
e.g.: the lines of code for the list function could be
if (msg.equals("list")) {
    printList();
}
To simplify the method in general and making it easier to read

LEESY02 commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,94 @@
+
+    public static void main(String[] args) {
+        Scanner msg = new Scanner(System.in);
+        System.out.println(
```
I think it would be easier to read if each of these print messages are encapsulated in a method, to make the main method shorter and more concise

LEESY02 replied:
for e.g.:  
public static void main(String[] args) {
        Scanner msg = new Scanner(System.in);
        printIntroMessage();
        .......
        printOutroMessage();
}

private static void printIntroMessage() {
        sout("This is an intro message :D");
}
private static void printOutroMessage() {
        sout("This is an outro message :D");
}

LEESY02 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,39 @@
+    private String desc;
+    private boolean isDone;
+
+    // Type 0: To-do (default)
```
Perhaps instead of using int type to identify the different sub tasks, could have Task as a parent class and the other subtasks as a children class

i.e.:
public class Task {
}
public class ToDo extends Task {
}

Using super classes' toString() in the child's classes' toString is an idea to automate the printing of each task
Task's toString: "[ ] Task A"
Child's toString: "[T]" + super.toString()
Of course this is a very simplified example, it is up to you on how you implement it

LEESY02 commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,94 @@
+import java.util.Scanner;
```
Overall it was a good read with all the functionalities, there is only a few things to touch up :D

LEESY02 commented on src/main/java/Baguette.java:
```diff
@@ -1,4 +1,4 @@
-public class Duke {
+public class Baguette {
```
Nice class name, oui oui baguette

LEESY02 reviewed (commented):
I think majority of the code looks good! I gave some minor edit suggestions, to give them some consideration :D

Rishi7830 commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,94 @@
+                msg + "\n" +
+                "-------------------------- \n");
+        Scanner newMsg = new Scanner(System.in);
+        echo(newMsg.nextLine());
```
This commented code looks like a code of the previous level so remove it :) for better code quality

Rishi7830 commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,94 @@
+            int type = 0; // defaults to to-do if no prefix
+            int index = 0;
+
+            if (msg.startsWith("todo ")) {
```
Good job on not using the arrowhead style code style

Rishi7830 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,39 @@
+    // Type 2: Event
+    private int type;
+
+    public Task(String description, int type) {
```
Good job on structuring the code logically

Rishi7830 commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,94 @@
+                System.out.println((i + 1) + ". " + list[i]);
+            }
+            System.out.println(DIVIDER + NL);
+        } else if (msg.startsWith("mark ")) {
```
Good job on not using the arrowhead style code

Rishi7830 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,39 @@
+
+    @Override
+    public String toString() {
+        String typeBox = new String("");
```
I like your code structure

Rishi7830 reviewed (commented):
Good Job

yangyi-zhu committed:
Restructured task types using inheritance and abstract classes

yangyi-zhu committed:
Added a help message to display when prefix is undefined

yangyi-zhu committed:
Added block preventing users from entering blank descriptions for to-dos
Removed redundant commented method from an earlier increment

yangyi-zhu committed:
Merge pull request #1 from yangyi-zhu/branch-Level-5

Added block preventing users from entering blank descriptions for to-dos

yangyi-zhu committed:
Modified help message

yangyi-zhu committed:
Divided deadline substring from message

yangyi-zhu committed:
Fixed program infinitely looping when an unexpected prefix is detected
Fixed displaying a false message about a null task being added when a list is requested

aditichadha1310 commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,110 @@
+    private static int count = 0;
+    private static final Task[] list = new Task[100];
+
+    public static void checkInput(String msg) {
```
Avoid lengthy methods. Keep the length of each method to about 30 lines of code.

aditichadha1310 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,30 @@
+
+    @Override
+    public String toString() {
+        return getType() + (isDone ? "[✓] " : "[ ] ") + desc;
```
Avoid usage of magic strings.

aditichadha1310 commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,110 @@
+                    list[count] = new Event(msg.substring(index), "", "");
+                } else {
+                    repeat = true;
+                    System.out.println(
+                            NL + DIVIDER + NL +
+                                    "The supported types and corresponding formats are as follows:" + NL +
+                                    "To-Dos: todo [description]" + NL +
+                                    "Deadlines: deadline [description] ddl: [DD-MM-YY HH:MM]" + NL +
+                                    "Events: event [description] from: [DD-MM-YY HH:MM] to: [DD-MM-YY HH:MM]" + NL +
+                                    NL + "Examples:" + NL +
+                                    "todo Do the laundry" + NL +
+                                    "deadline CS2113 Increment L5 ddl: 14-02-25 16:00" + NL +
+                                    "event Josh's birthday party from: 03-03-25 15:00 to: 04-03-25 00:00" +
+                                    NL + DIVIDER + NL
```
Since the print statements are common for many commands -  you can consider creating a function instead of repeating the same code lines for each command

yangyi-zhu committed:
Added option to delete tasks

yangyi-zhu committed:
Modularized essential functions for clarity and ease of updates
Replaced magic string messages with constants stored in a distinct utility class
Renamed Main file

yangyi-zhu committed:
Replaced magic numbers symbolizing indices to separate descriptions with constants
Renamed Messages to Constants

yangyi-zhu committed:
Added messages to delete function

yangyi-zhu committed:
Merge pull request #2 from yangyi-zhu/branch-Level-6

Branch level 6

yangyi-zhu committed:
Added logo display to the welcome message

yangyi-zhu committed:
Reorganized datatype classes into package

Added logo display to the welcome message

yangyi-zhu committed:
Restructured using Java ArrayList instead of a fixed-size array

yangyi-zhu committed:
Added ability to save list and load from file locally

yangyi-zhu committed:
Merge pull request #4 from yangyi-zhu/branch-Level-7

Added ability to save list and load from file locally

yangyi-zhu committed:
Added class for custom exceptions

yangyi-zhu committed:
Merge branch 'master' of https://github.com/yangyi-zhu/ip

yangyi-zhu committed:
Added class for custom exceptions

yangyi-zhu committed:
Updated add task logic with exceptions
Fixed problems with prefix length resulting in erroneous load ins

yangyi-zhu committed:
Further modularized code for ease of update

yangyi-zhu committed:
Fixed commands lacking spaces leading to fatal crashes

yangyi-zhu committed:
Added feature to search for tasks containing a keyword
Updated long strings to match coding standards

yangyi-zhu committed:
Merge pull request #5 from yangyi-zhu/branch-Level-9

Added feature to search for tasks containing a keyword

yangyi-zhu committed:
Updated Deadline and Event classes to use LocalDateTime instead of plain Strings

yangyi-zhu committed:
Merge pull request #6 from yangyi-zhu/branch-Level-8

Updated Deadline and Event classes to use LocalDateTime instead of pl…

yangyi-zhu committed:
Added JavaDoc comments
Fixed an error with events displaying the dates twice

yangyi-zhu committed:
Added User Guide

yangyi-zhu committed:
Update README.md

yangyi-zhu committed:
fix: Fixed LocalDateTime parsing error always returning current time
fix: Fixed marking and unmarking invalid indices crashing the program
fix: Completed and incomplete tasks can no longer be marked and unmarked respectively
refactor: Reduced nesting levels

yangyi-zhu committed:
Update README.md

yangyi-zhu committed:
Update README.md

yangyi-zhu committed:
Update README.md

yangyi-zhu committed:
Update README.md

yangyi-zhu committed:
Update README.md

yangyi-zhu committed:
Update README.md

yangyi-zhu committed:
Update README.md

yangyi-zhu committed:
Update README.md

yangyi-zhu committed:
Update README.md

yangyi-zhu committed:
Update README.md

yangyi-zhu committed:
Update README.md

yangyi-zhu committed:
Merge pull request #8 from yangyi-zhu/branch-UG

Branch ug

yangyi-zhu committed:
Update README.md
